sktime/transformations/series/simple_fourier.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+
+"""Simple Fourier transform for time series, interfacing numpy.fft"""
+
+__author__ = ["blazingbhavneek"]
+
+
+import numpy as np
+import pandas as pd
+from numpy.fft import fft, fftfreq
+from sktime.transformations.base import BaseTransformer
+
+
+class simple_fourier(BaseTransformer):
+    r"""Simple Fourier transform for time series
+
+    Fourier Transform is used to test data for periodicity.
+    Non-periodic/Noisy/Random data will have limited and
+    random amplitudes for frequencies, whereas periodic data
+    will have exceptional amplitudes for certain frequencies.
+
+    The implementation is based on the fast fourier transform from numpy.fft
+
+    Examples
+    --------
+    >>> from sktime.transformations.series.simple_fourier import simple_fourier
+    >>> from sktime.datasets import load_airline
+    >>> X = load_airline()
+    >>> transformer = simple_fourier()
+    >>> X_ft = transformer._transform(X)
+    """
+
+    _tags = {
+        "scitype:transform-input": "Series",
+        "scitype:transform-output": "Series",
+        "scitype:instancewise": True,
+        "scitype:transform-labels": "None",
+        "X_inner_mtype": "pd.Series",
+        "y_inner_mtype": "None",
+        "univariate-only": False,
+        "requires_y": False,
+        "fit_is_empty": True,
+        "capability:inverse_transform": False,
+        "capability:unequal_length": True,
+        "handles-missing-data": False,
+    }
+
+    def __init__(self):
+        super(simple_fourier, self).__init__()
+
+    def _transform(self, X):
+        """Transform X and return a transformed version.
+
+        private _transform containing core logic, called from transform
+
+        Parameters
+        ----------
+        X : Series mtype X_inner_mtype
+
+        Returns
+        -------
+        transformed version of X
+        """
+
+        X_sort = X.sort_index()  # In-case the data is unsorted
+
+        ser_size = X.size
+        sig = X_sort.to_numpy()
+
+        # numpy.fft methods
+        dft_seq = np.abs(fft(sig))
+        sample_freq = fftfreq(ser_size)
+
+        # Removing negative conjugate of frequncies
```

Comment:
Would using [`rfft`](https://numpy.org/doc/stable/reference/generated/numpy.fft.rfft.html#numpy.fft.rfft) instead of `fft` make this section of the code redundant and also provide a faster solution?

Reply:
Yes! According to [this](https://www.researchgate.net/figure/5-Speed-comparison-of-FFT-method-and-RFFT-method_tbl7_49909866) article its efficiency increases with dataset size, I will make changes right away!